File: software/examples/ruby/example_threshold.rb
#!/usr/bin/env ruby
# -*- ruby encoding: utf-8 -*-

require 'tinkerforge/ip_connection'
require 'tinkerforge/bricklet_analog_in_v2'

include Tinkerforge

HOST = 'localhost'
PORT = 4223
UID = 'ABC' # Change to your UID

ipcon = IPConnection.new # Create IP connection
ai = BrickletAnalogInV2.new UID, ipcon # Create device object

ipcon.connect HOST, PORT # Connect to brickd
# Don't use device before ipcon is connected

# Get threshold callbacks with a debounce time of 10 seconds (10000ms)
ai.set_debounce_period 10000

# Register threshold reached callback for voltage smaller than 5V
ai.register_callback(BrickletAnalogInV2::CALLBACK_VOLTAGE_REACHED) do |voltage|
  puts "Voltage dropped below 5V: #{voltage/1000.0}"
end

# Configure threshold for "smaller than 5V" (unit is mV)
ai.set_voltage_callback_threshold '<', 5*1000, 0

puts 'Press key to exit'
$stdin.gets
ipcon.disconnect

File: software/examples/go/example_threshold.go
package main

import (
	"fmt"
	"github.com/Tinkerforge/go-api-bindings/analog_in_v2_bricklet"
	"github.com/Tinkerforge/go-api-bindings/ipconnection"
)

const ADDR string = "localhost:4223"
const UID string = "XYZ" // Change XYZ to the UID of your Analog In Bricklet 2.0.

func main() {
	ipcon := ipconnection.New()
	defer ipcon.Close()
	ai, _ := analog_in_v2_bricklet.New(UID, &ipcon) // Create device object.

	ipcon.Connect(ADDR) // Connect to brickd.
	defer ipcon.Disconnect()
	// Don't use device before ipcon is connected.

	// Get threshold receivers with a debounce time of 10 seconds (10000ms).
	ai.SetDebouncePeriod(10000)

	ai.RegisterVoltageReachedCallback(func(voltage uint16) {
		fmt.Printf("Voltage: %f V\n", float64(voltage)/1000.0)
	})

	// Configure threshold for voltage "smaller than 5 V".
	ai.SetVoltageCallbackThreshold('<', 5*1000, 0)

	fmt.Print("Press enter to exit.")
	fmt.Scanln()
}
